edxadapt/static/js/src/edxadapt.js

```diff
@@ -65,8 +65,8 @@ function EdxAdaptXBlock(runtime, element) {
         })
         .fail(function(jqXHR, textStatus) {
             if (jqXHR.status == 404) {
-                // Student not found in EdxAdapt. Let's register them
-                registerUser();
+                // Student not found in EdxAdapt. Let's register him
+                $.when(configureUser()).then(registerUser());
```

Comment:
Yes `$.ajax()` returns promise subclass, but `configureUser()` does not return it.

So `$.when()` receives `undefined` as an input. And if I read jQuery docs correctly, following is going to happen:
> If a single argument is passed to `jQuery.when()` and it is not a Deferred or a Promise, it will be treated as a resolved Deferred and any doneCallbacks attached will be executed immediately.